back/engines/commercial/user_custom_fields/app/controllers/user_custom_fields/web_api/v1/ref_distributions_controller.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module UserCustomFields
+  module WebApi::V1
+    class RefDistributionsController < ::ApplicationController
+      def show
+        render json: serialize(reference_distribution)
+      end
+
+      def create
+        ref_distribution = authorize(Representativeness::RefDistribution.new(create_params))
+
+        Representativeness::RefDistribution.transaction do
+          remove_reference_distribution_if_any(params[:custom_field_id])
+          ref_distribution.save!
+        end
+
+        side_fx.after_create(ref_distribution, current_user)
+        render json: serialize(ref_distribution), status: :created
+      rescue ActiveRecord::RecordInvalid => e
+        render json: { errors: e.record.errors.full_messages }, status: :unprocessable_entity
+      end
+
+      def destroy
+        reference_distribution.destroy!
+        side_fx.after_destroy(reference_distribution, current_user)
+        head :no_content
+      end
+
+      private
+
+      def remove_reference_distribution_if_any(custom_field_id)
+        Representativeness::RefDistribution.where(
+          custom_field_id: custom_field_id
+        ).destroy_all
+      end
+
+      def side_fx
+        Representativeness::SideFxRefDistributionService.new
```

Comment:
Consider memoizing the result, because this method will be called twice from actions with `before_xxx` and `after_xxx`.

Reply:
Well, indeed if I add a call to `before_xxx`. In any case, I'll add it as it's low cost and makes the code more future-proof.